pkg/streamingpromql/testdata/ours/aggregators.test

```diff
@@ -31,3 +31,41 @@ load 1m
 eval range from 1m to 1m30s step 1s sum(some_metric_with_staleness)
   # Should return no results.
 
+clear
+
+# Test native histogram aggregations
+load 1m
+	single_histogram{label="value"} {{schema:0 sum:2 count:4 buckets:[1 2 1]}} {{sum:2 count:4 buckets:[1 2 1]}}
+	single_histogram{label="value2"} {{schema:1 sum:5 count:5 buckets:[1 3 1]}} {{schema:0 sum:2 count:4 buckets:[1 2 1]}}
+	single_histogram{label="value2", another="value"} {{schema:1 sum:10 count:9 buckets:[3 3 3]}} {{schema:2 sum:2 count:4 buckets:[1 2 1]}}
+
+eval instant at 0m sum(single_histogram)
+	{} {{schema:0 sum:17 count:18 buckets:[5 12 1]}}
+
+eval instant at 0m sum by (label) (single_histogram)
+	{label="value"} {{schema:0 count:4 sum:2 buckets:[1 2 1]}}
+	{label="value2"} {{schema:1 count:14 sum:15 buckets:[4 6 4]}}
+
+eval range from 0 to 1m step 1m  sum(single_histogram)
+	{} {{schema:0 sum:17 count:18 buckets:[5 12 1]}} {{schema:0 sum:6 count:12 buckets:[3 7 2]}}
+
+clear
+
+# Test a mix of float and histogram values
+load 1m
+	single_histogram{label="value"}                    0 1 {{schema:0 sum:2 count:4 buckets:[1 2 1]}}  {{sum:2 count:4 buckets:[1 2 1]}}          2
+	single_histogram{label="value2"}                   0 2 {{schema:1 sum:5 count:5 buckets:[1 3 1]}}  {{schema:0 sum:2 count:4 buckets:[1 2 1]}} 4
+	single_histogram{label="value2", another="value"}  0 3 {{schema:1 sum:10 count:9 buckets:[3 3 3]}} {{schema:2 sum:2 count:4 buckets:[1 2 1]}} 6
```

Comment:
Just thought of another test case that would be good to add: `sum` where some series in the one group have floats and some have native histograms at the same time step

Reply:
I think I tried to do that, but couldn't find a way with the promql test framework to do so. I might need to write some custom tests for these mixed edge cases.

Reply:
The old code still has a check:
```
	pointCount := 0
	for _, p := range thisGroup.present {
		if p {
			pointCount++
```
So not sure if this would be any worse.

Nonetheless, I've opened a separate PR for this here along with a benchmark: https://github.com/grafana/mimir/pull/8442

I'm going to resolve this and move the conversation over there.